docs/datetime.txt

```diff
@@ -228,55 +227,52 @@ For example, lets take a timezone as defined by the following class:
   
   class ATZInfo(tzinfo):
 
-     def tzname(self, dt):
+      def tzname(self, dt):
           return 'A TimeZone'
         
-     def utcoffset(self, dt):
-          # In general, this timezone is 5 hours behind UTC
-          offset  = timedelta(hours=-5)
-          return offset+self.dst(dt)
-   
-     def dst(self, dt):
-          # However, between March and September, it is only
-          # 4 hours behind UTC
-          if 3 < dt.month < 9:
-              return timedelta(hours=1)
```

Comment:
I think the dst stuff is important to include as it's where a lot of this complexity comes from.

Reply:
I specifically removed this because I don't think it is germain to explaining timezone treatment.

The main things to convey are that:
1. Even though test_datetime deals with naive datetimes, when required, these may be ***interpreted*** with a specific timezone.
2. The ***interpreted*** timezone defaults to UTC unless otherwise specified in the test_datetime() "constructor".
3. Externally observable behaviour of datetime APIs should then be as per stdlib.

I think the details about how complicated a timezone might or might not be are irrelevant to this understanding and just add further confusion to the examples.

Reply:
If we agree that the three points above are the essence of how things work I can have another crack at rewriting the documentation to focus on these three points. LMK

Reply:
1. Yeah, ish, the fundamental problem is what timezone the naive datetime stream in test_datetime is considered to be in. 

2. I don't remember this being the intention of the code, but it's been a long time, can you point me to the tests to back this up? (I'm not saying you're wrong, I genuinely can't remember and I'd like to be sure, since we're splitting hair here already).

3. I guess, but I'm not really sure what that means - some of the stdlib behaviour is expressed somewhat unclearly, *and* it's fundamentally different in that it's dealing with real localtime and utc, and at a system level these end up being horribly complicated in their own right. (eg: tell me programatically what timezone you local system is set to).

I disagree about timezone complexity, I think DST is to common and tricky that it's worth showing the difficulties it creates specifically in these examples. There are even more complicated timezones (half hour changes, random changes through history, etc) that aren't worth worrying about, but DST is horrible, hard and everyone needs to think about it when they're mocking dates and times.

Reply:
1. Sounds like we're saying the exact same thing?
2. I can't speak to the original _intentions_ of the code (that's the reason I raised this issue -- to establish if the current behaviour is in line with intentions, or a bug that's liable to be fixed) but that's precisely what [the implementation](https://github.com/Simplistix/testfixtures/blob/master/testfixtures/tdatetime.py#L70) does.
If "having `now()` and `utcnow()` return the same thing in simple cases where you're not thinking about timezones" is part of the intention, then I think this directly implies the intention that "The interpreted timezone defaults to UTC unless otherwise specified in the test_datetime() "constructor"."
3. I think the complications you talk about are orthogonal. Python has a concept of system timezone, regardless of how complicated the implementation of retrieving that is.

Although it might be good to highlight the complications of DST transitions as one particularly common nuance of timezones, I think that should be done separately in order to avoid complicating the explanation of how testfixtures works.